source/EngineTests/PreviewDescriptionConverterServiceTests.cpp

```diff
@@ -0,0 +1,186 @@
+#include <gtest/gtest.h>
+
+#include "Base/Definitions.h"
+#include "EngineInterface/Descriptions.h"
+#include "EngineInterface/PreviewDescriptionConverterService.h"
+#include "EngineInterface/DescriptionEditService.h"
+#include "IntegrationTestFramework.h"
+
+class PreviewDescriptionConverterServiceTests 
+    : public IntegrationTestFramework
+{
+public:
+    PreviewDescriptionConverterServiceTests()
+        : IntegrationTestFramework(std::nullopt, {100, 100})
+    {}
+    virtual ~PreviewDescriptionConverterServiceTests() = default;
+
+protected:
+    CellDescription createCell(uint64_t id, RealVector2D pos, int color = 0, uint16_t nodeIndex = 0)
+    {
+        return CellDescription().id(id).pos(pos).color(color).genomeNodeIndex(nodeIndex);
+    }
+};
+
+TEST_F(PreviewDescriptionConverterServiceTests, convertEmptyCollection)
+{
+    CollectionDescription input;
+    
+    auto result = PreviewDescriptionConverterService::get().convert(input);
+    
+    EXPECT_TRUE(result._cells.empty());
+    EXPECT_TRUE(result._connections.empty());
+}
+
+TEST_F(PreviewDescriptionConverterServiceTests, convertSingleCell)
+{
+    CollectionDescription input;
+    auto cell = createCell(1, {10.0f, 20.0f}, 3, 5);
+    input.addCell(cell);
+    
+    auto result = PreviewDescriptionConverterService::get().convert(input);
+    
+    EXPECT_EQ(1, result._cells.size());
+    EXPECT_EQ(0, result._connections.size());
+    
+    EXPECT_EQ(3, result._cells[0]._color);
+    EXPECT_EQ(5, result._cells[0]._nodeIndex);
+    
+    EXPECT_FLOAT_EQ(0.0f, result._cells[0]._pos.x);
+    EXPECT_FLOAT_EQ(0.0f, result._cells[0]._pos.y);
+}
+
+TEST_F(PreviewDescriptionConverterServiceTests, convertMultipleCells)
+{
+    CollectionDescription input;
+    
+    auto cell1 = createCell(1, {10.0f, 10.0f}, 1, 2);
+    auto cell2 = createCell(2, {20.0f, 10.0f}, 2, 3);
+    auto cell3 = createCell(3, {15.0f, 20.0f}, 3, 4);
+    
+    input.addCell(cell1);
+    input.addCell(cell2);
+    input.addCell(cell3);
+    
+    auto result = PreviewDescriptionConverterService::get().convert(input);
+    
+    EXPECT_EQ(3, result._cells.size());
+    EXPECT_EQ(0, result._connections.size());
+    
+    for (int i = 0; i < 3; ++i) {
+        EXPECT_EQ(i + 1, result._cells[i]._color);
+        EXPECT_EQ(i + 2, result._cells[i]._nodeIndex);
+    }
+    
+    RealVector2D center = {0.0f, 0.0f};
```

Comment:
Use DescriptionEditService::calcCenter instead of calculating the center yourself.